README.md

```diff
@@ -56,94 +56,56 @@ jobs:
       - uses: haskell-actions/hlint-run@v2
 ```
 
+<!-- action-docs-inputs action="action.yml" -->
 ## Inputs
 
-All inputs are optional.
-
-- `working-directory`: working directory for all `run` steps.
-
-  Default is `.`. Useful for a multi-project repository.
-
-- `stack-yaml`: path to use instead of `stack.yaml`.
-
-  Default is `stack.yaml`. Expected to be relative to `working-directory`.
-
-- `test`: whether tests should be executed
-
-  Default `true`.
-
-- `stack-arguments`: global arguments for **all** `stack` invocations
-
-  Default is `--no-terminal --stack-yaml {stack-yaml}`, and if `stack-yaml` is
-  the string `"stack-nightly.yaml"`, `--resolver nightly` will be added.
-
-- `stack-build-arguments`: arguments for **all** `stack build` invocations
-
-  Default is `--fast --pedantic`. If you are building executables, you probably
-  want to override this to remove `--fast`.
-
-- `stack-build-arguments-dependencies`: additional arguments for `stack build`
-  in the _Dependencies_ step.
-
-- `stack-build-arguments-build`: additional arguments for `stack build` in the
-  _Build_ step.
-
-- `stack-build-arguments-test`: additional arguments for `stack build` in the
-  _Test_ step.
-
-- `stack-setup-arguments`: additional arguments for `stack setup`
-
-  Default is none.
-
-- `stack-query-arguments`: additional arguments for `stack query`
-
-  Default is none.
-
-- `stack-path-arguments`: additional arguments for `stack path`
-
-  Default is none.
-
-- `cache-prefix`: prefix applied to all cache keys. This can be any value you
-  like, but teams often use `v{N}` and bump it to `v{N+1}` when/if they need to
-  explicitly bust caches. The default is empty.
-
-- `cache-save-always`: save artifacts to the cache even if the build fails.
-  This may speed up builds in subsequent runs at the expense of slightly-longer
-  builds when a full cache-hit occurs. (Since `@v4.2.0`)
-
+| name | description | required | default |
```

Comment:
One think I noticed when I was working on this myself is that a value with a `default` is not "required", even though we put `required: true` in the `action.yml`. Ideally, that would be reflected here by setting the table cell to something like `required && default != null`.

Have you thought about that at all?

Reply:
No, I haven't. Could you please provide an example of such a bad conversion?

So, the truth table will look like this? Why should it look so?

| required | default != null | final required
|---|---|---|
| true | true | true |
| true | false | true |
| false | true | false |
| false | false | false |

Reply:
> Could you please provide an example of such a bad conversion?

I wouldn't call it a "bad conversion", but every single input in the table you've generated here says `true` in the required column -- even though none of them are _actually_ required (because they have defaults).

> I think the current version where values don't depend on each other is more straightforward

Yeah I agree. If you have nothing clever, I will probably just start removing the `required` key from any inputs that have defaults. There was a reason I started doing it that way, but now I can't remember.

Reply:
> If you have nothing clever,

Yes, I don't have anything.